src/oscar/apps/search/search_handlers.py

```diff
@@ -9,7 +9,43 @@
 FacetMunger = get_class('search.facets', 'FacetMunger')
 
 
-class SearchHandler(object):
+class SearchResultsPaginationMixin(object):
+    paginate_by = None
+    paginator_class = Paginator
+    page_kwarg = 'page'
+
+    def get_paginate_by(self, *args, **kwargs):
```

Comment:
This method is not passed any arguments from where we are calling it?

Reply:
Do we need this method at all? If someone wanted to override with custom logic they could just define a `paginate_by` property.

Reply:
Agreed. Removed.